server/index.js

```diff
@@ -2,13 +2,51 @@ require('dotenv/config');
 const express = require('express');
 const staticMiddleware = require('./static-middleware');
 const errorMiddleware = require('./error-middleware');
+const fetch = require('node-fetch');
 
 const app = express();
 
 app.use(staticMiddleware);
 
-app.get('/api/hello', (req, res) => {
-  res.json({ hello: 'world' });
+app.get('/api/discover', (req, res) => {
+  let token;
+  let tokenType;
+  let expires;
+  if (!expires || expires - new Date().getTime() < 1) {
+    getOAuth()
+      .then(() => getDoggo());
+  } else {
+    getDoggo();
+  }
+
+  function getOAuth() {
+    return fetch('https://api.petfinder.com/v2/oauth2/token', {
+      method: 'POST',
+      body: 'grant_type=client_credentials&client_id=' + process.env.PETFINDER_API_KEY + '&client_secret=' + process.env.PETFINDER_API_SECRET,
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded'
+      }
+    })
+      .then(res => res.json())
+      .then(data => {
+        token = data.access_token;
+        tokenType = data.token_type;
+        expires = new Date().getTime() + (data.expires_in * 1000);
+      })
+      .catch(err => console.error('Fetch failed at getOAuth', err));
```

Comment:
We always pass errors to `next()` in our express code. However, I recommend moving it to your `getDoggo()` and `getOAuth()` chains.